plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchClientOptions.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { ConnectionOptions as TLSConnectionOptions } from 'tls';
+
+/**
+ * Options used to configure the `@elastic/elasticsearch` client and
+ * are what will be passed as an argument to the
+ * {@link ElasticSearchEngine.newClient} method
+ *
+ * They are drawn from the `ClientOptions` class of `@elastic/elasticsearch`,
+ * but are maintained separately so that this interface is not coupled to
+ */
+export interface ElasticSearchClientOptions {
+  provider?: 'aws' | 'elastic';
```

Comment:
The drawback being, now you have to maintain this one as it drifts away from the actual client when it gets upgraded. But I can see the tradeoffs

Reply:
@freben Yeah, it's a tough one.